app/src/main/java/ch/nevis/exampleapp/coroutines/domain/interaction/AuthenticationAuthenticatorSelectorImpl.kt

```diff
@@ -41,17 +50,38 @@ class AuthenticationAuthenticatorSelectorImpl(
         val cancellableContinuation =
             operationState.cancellableContinuation ?: throw BusinessException.invalidState()
 
-        val authenticators = authenticatorSelectionContext.authenticators().filter {
-            it.registration().isRegistered(authenticatorSelectionContext.account().username()) &&
-                    it.isSupportedByHardware
+        val authenticatorItems = authenticatorSelectionContext.authenticators().mapNotNull {
+            mapForAuthentication(it, authenticatorSelectionContext)
         }.toSet()
 
         operationState.username = authenticatorSelectionContext.account().username()
         operationState.authenticatorSelectionHandler = authenticatorSelectionHandler
 
         cancellableContinuation.resume(
-            SelectAuthenticatorResponse(authenticators)
+            SelectAuthenticatorResponse(authenticatorItems)
         )
     }
     //endregion
+
+    //region Private Interface
+    private fun mapForAuthentication(
+        authenticator: Authenticator,
+        context: AuthenticatorSelectionContext
+    ): AuthenticatorItem? {
+        return if (authenticator.registration()
+                .isRegistered(context.account().username()) && authenticator.isSupportedByHardware
```

Comment:
Would be good to have a log statement here, this way, the devs can pinpoint where the check happens.

Reply:
Fixed. I also added it to simplified example app. https://github.com/nevissecurity/nevis-mobile-authentication-sdk-example-app-android/pull/5